app/controllers/shelters_controller.rb

```diff
@@ -23,10 +28,17 @@ def edit
   def update
     @shelter = Shelter.find(params[:id])
     @shelter.update(shelter_params)
-    redirect_to "/shelters/#{@shelter.id}"
+    if @shelter.save
```

Comment:
Nice adding in those edge cases. I think we could probably replace `@shelter.save` with `@shelter.update(shelter_params)` and then remove that line of code from above